docs/osf/addons.rst

```diff
@@ -31,31 +32,228 @@ During your installation you created a virtual environment for OSF. Switch to th
+   -  EX: The wiki defines both a ``page`` view and a ``widget`` view
+
+-  ``CATEGORIES`` (list of strings)
+
+   -  A list of categories this add-on should be displayed under when
+      the user is “browsing” add-ons
+      
+   - SHOULD be one of ``documentation``, ``storage``, ``citations``, ``security``, ``bibliography``, and ``other``
+   
+       - Additional categories can be added to ``ADDON_CATEGORIES`` in ``website.settings.defaults``
+
+-  ``INCLUDE_JS`` and ``INCLUDE_CSS``
+
+   -  Deprecated field, define as empty dict (``{}``)
+
+Optional Fields
+---------------
+
+Your add-on MAY define the following fields
+
+-  ``HAS_HGRID_FILES`` (boolean)
+-  A boolean that indicated that this add-on’s ``GET_HGRID_DATA``
+   function should be used to populate the files grid
+-  ``GET_HGRID_DATA`` (function)
+-  A function that returns HGrid/Treebeard formatted data to be included
+   in a project’s files grid
+-  ``USER_SETTINGS_MODEL`` (`StoredObject`_)
+-  MUST inherit from ``website.addons.base.AddonUserSettingsBase``
+-  A model that will be used to store settings for users
+-  Will be returned when ``User.get_addon('YourAddon')`` is called
+-  EX:
+
+   -  S3’s User settings is used to store the user’s AWS access keys
+
+-  ``NODE_SETTINGS_MODEL`` (`StoredObject`_)
+-  MUST inherit from ``website.addons.base.AddonNodeSettingsBase``
+-  A model that will be used to store settings for nodes
+-  Will be returned when ``Node.get_addon('YourAddon')`` is called
+-  ``NODE_SETTINGS_TEMPLATE`` (string to directory)
+-  A `mako`_ template for configuring your add-on’s node settings object
+-  ``USER_SETTINGS_TEMPLATE`` (string to directory)
+-  A `mako`_ template for configuring your add-on’s user settings object
+-  ``MAX_FILE_SIZE``
+-  This maximum size, in MB, that can be uploaded to your add-on, supposing it supports files
+
+
+Addon Structure
+------------------
+
+An add-on SHOULD have the following folder structure
 
 ::
 
-    $ cookiecutter https://github.com/chrisseto/osf-Addon-cookiecutter.git
+    website/addons/addonshortname/
+    ├── __init__.py
+    ├── model.py
+    ├── requirements.txt
+    ├── routes.py
+    ├── settings
+    │   ├── __init__.py
+    │   └── defaults.py
+    ├── static
+    │   ├── comicon.png
+    │   ├── node-cfg.js*
+    │   ├── tests
+    │   │   └── ...
+    │   └── user-cfg.js*
+    ├── templates
+    │   ├── log_templates.mako
+    │   ├── addonshortname_node_settings.mako*
+    │   └── addonshortname_user_settings.mako*
+    ├── tests
+    │   ├── __init__.py
+    │   ├── test_model.py
+    │   └── test_views.py
+    └── views
+        └── ...
+
+\* optional
+
+StoredObject
+============
+
+A ``StoredObject`` is a class from our package ``modularodm`` that
+represents an object stored in a database.
+
+For our use case this database will always be TokuMX, a fork of mongodb
+that provides transactions and performance increases.
+
+Defining a ``StoredObject`` is very similar to the syntax used in
```

Comment:
This is arguable. In general, this section might be too pedantic. I think it's enough to say:

```
StoredObject
==========

All models should be defined as subclasses of `framework.mongo.StoredObject`.